src/main/java/com/tfgm/services/TramStopGraphService.java

```diff
@@ -0,0 +1,87 @@
+package com.tfgm.services;
+
+import com.tfgm.models.Tram;
+import com.tfgm.models.TramStop;
+import com.tfgm.models.TramStopContainer;
+import java.util.Arrays;
+import java.util.Queue;
+
+public class TramStopGraphService {
+
+  public void tramDeparture(String endOfLine, TramStop tramStop) {
+    Tram departingTram;
+    Queue<Tram> tramQueue = tramStop.getTramQueue();
+
+    if (tramQueue.size() > 0) {
+      departingTram = tramQueue.remove();
+    } else {
+      departingTram = new Tram(endOfLine.length() * tramStop.getStopName().length(), endOfLine);
+    }
+
+    TramStopContainer[] nextStops = tramStop.getNextStops();
+
+    for (TramStopContainer tramStopContainer : nextStops) {
+
+      if (!(tramStopContainer.getTramStop().getStopName().equals("Exchange Square"))
+          || endOfLine.matches("East Didsbury|Shaw and Crompton|Rochdale Town Centre")) {
+
+        // NB: Due to the nature of the recursive algorithim the final destination of Eccles via
+        // MediaCityUK and Ashton via MCUK needs to be changed to MediaCityUK so the tramstop can be
+        // found.
+        if (findEndOfLine(
+            Arrays.stream(departingTram.getDestination().split(" "))
+                    .reduce((first, second) -> second)
+                    .get()
+                    .matches("MCUK|MediaCityUK")
+                ? "MediaCityUK"
+                : departingTram.getDestination(),
+            tramStopContainer.getTramStop())) {
+          tramStopContainer.getTramLinkStop().addTram(departingTram);
+          System.out.println(
```

Comment:
This looks better, so now instead of printing the result return it as a String and then in your controller you can return the data as a ResponseEntity to the client.